### PR TITLE
go: update arm version to 1.16rc1

### PR DESCRIPTION
**Pull request by christophwitzko**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/go.rb [hunk 1]
```diff
@@ -91,8 +91,8 @@ def caveats
 
     if Hardware::CPU.arm?
       s += <<~EOS
-        This is a beta version of the Go compiler for Apple Silicon
-        (Go 1.16beta1).
+        This is a release candidate version of the Go compiler for Apple Silicon
+        (Go go1.16rc1).
```
Comment (carlocab) [inline Formula/go.rb:95]: Should it just be `Go 1.16rc1` or `go1.16rc1`, and not a mix of both?

Comment (christophwitzko) [reply Formula/go.rb:95]: Oh yeah, right 😅

File: Formula/go.rb [hunk 2]
```diff
@@ -5,8 +5,8 @@ class Go < Formula
 
   stable do
     if Hardware::CPU.arm?
-      url "https://golang.org/dl/go1.16beta1.src.tar.gz"
-      sha256 "48e032c8cf71af4dc8119a29ee829c4fbd5265e32fd012564d4a70bb207695c1"
+      url "https://golang.org/dl/go1.16rc1.src.tar.gz"
+      sha256 "6a33569f9d0d21db31614086cc2a4f0fbc683b41c1c53fb512a1341ce5763ff5"
       version "1.15.7"
```
Comment (carlocab) [inline Formula/go.rb:10]: You probably need to add `revision 1` to the end of this `if` block, or else anyone who has `go` installed won't get the updated version unless they do `brew reinstall go`.

Check this with `brew audit --strict go` first, though; I might be mistaken about where the `revision` line needs to go here.

**Review by carlocab — COMMENTED** [review]
Thanks, @christophwitzko.

File: Formula/go.rb (hunk 2 above)
Comment (christophwitzko) [reply Formula/go.rb:10]: If I add the revision I get the error "Error: No available formula or cask with the name" 🤔

Comment (carlocab) [reply Formula/go.rb:10]: Oof. Ok. Try moving the `revision 1` line just outside the `stable` block. Or just basically put it somewhere just before the install `block`. `brew audit --strict --fix go` should hopefully help you put it where it belongs.

Comment (christophwitzko) [reply Formula/go.rb:11]: okay, that seems to work. Thank you!

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
The bottles look to have built fine, plus this PR only changes things on ARM anyway. `brew pr-pull` isn't working for me for some reason though... CC @fxcoudert

**Comment by fxcoudert** [comment]
OK I'm going a manual merge then

**Comment by christophwitzko** [comment]
Thank you @carlocab @fxcoudert 🎉

**Pull request closed without merging**